### ISSUE TITLE
Change Request: Allow ignore-path to be set permanently

### ISSUE THREAD

**Issue by leonheess** [eslint/eslint#15848]
### What problem do you want to solve?

The CLI already allows to use an alternate file like .gitignore to be used to ignore files and dirs ([docs](https://eslint.org/docs/user-guide/configuring/ignoring-code#using-an-alternate-file)). However, you cannot specify this in any configuration so no IDE is able to pick up on it.

### What do you think is the correct solution?

The setting of an ignore path should also be possible via the configuration like this

```
{
    "ignorePath": "./.gitignore",
    "rules": {
        //...
    }
}
```

### Additional comments

Related: #3529

**Comment by mdjermanovic** [comment]
Hi @leonheess, thanks for the issue!

ESLint also supports [`ignorePatterns` in Config Files](https://eslint.org/docs/user-guide/configuring/ignoring-code#ignorepatterns-in-config-files). You can load the alternate ignore file in the config module and use its content in `ignorePatterns`:

```js
module.exports = {
    ignorePatterns: require("fs").readFileSync("./.gitignore", "utf8").split("\n"),
    rules: {
        // ...
    }
};
```

I can see how the proposed new configuration option would be useful, but since there is a relatively easy way to achieve this functionality with the existing options, a new option might be unnecessary.

**Comment by nzakas** [comment]
The current config system is frozen and so we aren’t making any further changes. The new config system doesn’t load anything for you, so it would be up to you to manually load ignore patterns from wherever you want.
